File: python-algorithm/leetcode/problem_1460.py
"""1460. Make Two Arrays Equal by Reversing Sub-arrays
https://leetcode.com/problems/make-two-arrays-equal-by-reversing-sub-arrays/
"""
from collections import Counter
from typing import List


class Solution:
    def canBeEqual(self, target: List[int], arr: List[int]) -> bool:
        counter1, counter2 = Counter(arr), Counter(target)
        for k, v in counter1.items():
            if counter2[k] != v:
                return False
        return True

File: java-algo/src/main/java/com/leetcode/Problem1460.java
package com.leetcode;

import java.util.Arrays;

/**
 * 1460. Make Two Arrays Equal by Reversing Subarrays
 * https://leetcode.com/problems/make-two-arrays-equal-by-reversing-subarrays/
 */
public class Problem1460 {
    public boolean canBeEqual(int[] target, int[] arr) {
        Arrays.sort(target);
        Arrays.sort(arr);
        for (int i = 0; i < target.length; i++) {
            if (target[i] != arr[i]) {
                return false;
            }
        }
        return true;
    }
}
